test/gtest/ucp/test_ucp_wireup.cc

```diff
@@ -1769,14 +1769,14 @@ class test_ucp_address_v2 : public test_ucp_wireup {
         ucp_rsc_index_t rsc_index;
 
         for (rsc_index = 0; rsc_index < context->num_tls; ++rsc_index) {
-            if ((context->tl_rscs[rsc_index].md_index == ae->md_index) &&
+            if ((context->tl_rscs[rsc_index].dev_index == ae->dev_index) &&
```

Comment:
dev_iondex in in a different space in context and unpacked address, so cannot compare them
need to use uct_iface_is_reachable_v2 with SAME_DEVICE scope